arch/x86/adler32_avx512.c

```
@@ -91,12 +91,72 @@ Z_FORCEINLINE static uint32_t adler32_copy_impl(uint32_t adler, uint8_t *dst, co
+
+        vs3 = _mm512_slli_epi32(vs3, 6);
+        vs2 = _mm512_add_epi32(vs2, vs3);
+
+        adler0 = partial_hsum(vs1) % BASE;
+        adler1 = _mm512_reduce_add_epu32(vs2) % BASE;
+    }
+
+    return adler0 | (adler1 << 16);
+}
+
 Z_INTERNAL uint32_t adler32_avx512(uint32_t adler, const uint8_t *src, size_t len) {
```

Cppcheck (2.21.0)

[style] The function 'adler32_avx512' is never used.

(unusedFunction)